clients/spark-extensions/src/test/java/org/projectnessie/spark/extensions/ITNessieStatements.java

```diff
@@ -99,8 +100,50 @@ void testCreateBranchInExists() throws NessieNotFoundException {
   }
 
   @Test
-  void testCreateBranchIn() throws NessieNotFoundException {
+  public void testRefreshAfterMergeWithIcebergTableCaching() {
+    List<Object[]> result = sql("CREATE BRANCH %s IN nessie", refName);
+    assertEquals("created branch", row("Branch", refName, hash), result);
+    assertThat(sql("USE REFERENCE %s IN nessie", refName))
+        .hasSize(1)
+        .containsExactly(row("Branch", refName, hash));
+
+    sql("CREATE TABLE nessie.tbl (id int, name string)");
+    sql("INSERT INTO nessie.tbl select 23, \"test\"");
+    assertThat(sql("SELECT * FROM nessie.tbl")).hasSize(1).containsExactly(row(23, "test"));
+
+    sql(String.format("MERGE BRANCH %s INTO main in nessie", refName));
+    assertThat(sql("SELECT * FROM nessie.`tbl@main`")).hasSize(1).containsExactly(row(23, "test"));
+    assertThat(sql("SELECT * FROM nessie.`tbl@%s`", refName))
+        .hasSize(1)
+        .containsExactly(row(23, "test"));
+    sql("INSERT INTO nessie.tbl select 24, \"test24\"");
+    assertThat(sql("SELECT * FROM nessie.`tbl@main`")).hasSize(1).containsExactly(row(23, "test"));
+    assertThat(sql("SELECT * FROM nessie.tbl"))
+        .hasSize(2)
+        .containsExactlyInAnyOrder(row(23, "test"), row(24, "test24"));
+
+    // this still sees old data because tbl@testBranch is being cached in Iceberg
+    // due to "spark.sql.catalog.nessie.cache-enabled=true" by default
+    assertThat(sql("SELECT * FROM nessie.`tbl@%s`", refName))
+        .hasSize(1)
+        .containsExactly(row(23, "test"));
+
+    // disabling caching sees the correct data now because Iceberg doesn't cache the table anymore
+    assertThat(sqlWithDisabledCaching("SELECT * FROM nessie.`tbl@%s`", refName))
+        .hasSize(2)
+        .containsExactlyInAnyOrder(row(23, "test"), row(24, "test24"));
+  }
 
+  private static List<Object[]> sqlWithDisabledCaching(String query, Object... args) {
+    try (SparkSession sparkWithDisabledCaching = spark.newSession()) {
+      sparkWithDisabledCaching.conf().set("spark.sql.catalog.nessie.cache-enabled", "false");
+      List<Row> rows = sparkWithDisabledCaching.sql(String.format(query, args)).collectAsList();
+      return rows.stream().map(AbstractSparkTest::toJava).collect(Collectors.toList());
+    }
```

Comment:
I'm not sure this actually validates the effect of disabled caching in a session, because this is a new session. Will it not have an empty cache to begin with?

Reply:
you're right. Turns out this is actually a bit more difficult to test than I thought. I changed the test to just use a fresh session with an empty cache